### PR TITLE
markPaid: concatenate strings instead of override

### PR DESCRIPTION
**Pull request by waihin26**
Markpaid now concatenates Strings. UI also displays months in a sorted order.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -71,8 +73,8 @@ public ClassId getClassId() {
      * Returns an immutable monthPaid set, which throws {@code UnsupportedOperationException}
      * if modification is attempted.
      */
-    public Set<MonthPaid> getMonthsPaid() {
-        return Collections.unmodifiableSet(monthsPaid);
+    public SortedSet<MonthPaid> getMonthsPaid() {
```
Comment (hooyushing) [inline src/main/java/seedu/address/model/person/Person.java:76]: Good implementation of sort

File: src/main/java/seedu/address/model/person/MonthPaid.java [hunk 2]
```diff
@@ -45,4 +45,24 @@ public boolean equals(Object other) {
     public int hashCode() {
         return value.hashCode();
     }
+
+    @Override
+    public int compareTo(MonthPaid other) {
+        // Split the value into year and month
+        String[] thisParts = this.value.split("-");
```
Comment (hooyushing) [inline src/main/java/seedu/address/model/person/MonthPaid.java:52]: Maybe could name variable a little clearer

File: src/main/java/seedu/address/model/person/MonthPaid.java [hunk 3]
```diff
@@ -45,4 +45,24 @@ public boolean equals(Object other) {
...
+        int thisYear = Integer.parseInt(thisParts[0]);
+        int otherYear = Integer.parseInt(otherParts[0]);
+
+        // Compare by year first (descending order)
+        if (thisYear != otherYear) {
+            return Integer.compare(otherYear, thisYear);
+        }
+
+        // Compare by month if years are the same (descending order)
+        int thisMonth = Integer.parseInt(thisParts[1]);
```
Comment (hooyushing) [inline src/main/java/seedu/address/model/person/MonthPaid.java:64]: Clear on how the year is compared first followed by month

**Review by hooyushing — COMMENTED** [review]
LGTM except minor changes

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java [hunk 4]
```diff
@@ -38,7 +39,7 @@ public class MarkPaidCommand extends Command {
             + PREFIX_MONTHPAID + "2024-01"
             + PREFIX_MONTHPAID + "2024-02";
 
-    public static final String MESSAGE_MARKPAID_PERSON_SUCCESS = "Marked Person: %1$s";
+    public static final String MESSAGE_MARKPAID_PERSON_SUCCESS = "Markedpaid: %1$s";
```
Comment (sumomomomomo) [inline src/main/java/seedu/address/logic/commands/MarkPaidCommand.java:42]: nit: `Marked person as paid: %1$s` makes more sense

File: src/main/java/seedu/address/model/person/MonthPaid.java (hunk 2 above)
Comment (waihin26) [reply src/main/java/seedu/address/model/person/MonthPaid.java]: Updated with clearer names

File: src/main/java/seedu/address/logic/Messages.java [hunk 5]
```diff
@@ -47,5 +47,15 @@ public static String format(Person person) {
...
         return builder.toString();
     }
+    /**
+     * Formats the {@code person} for display to the user when marking paid
+     */
+    public static String markPaidFormat(Person person) {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(person.getName())
+                .append("; MonthsPaid: ");
+        person.getMonthsPaid().forEach(month -> builder.append(month).append(" "));
```
Comment (sumomomomomo) [inline src/main/java/seedu/address/logic/Messages.java:57]: Any reason why `builder::append` wasn't used? Referring to implementation found in `format`

File: src/main/java/seedu/address/logic/commands/MarkPaidCommand.java [hunk 6]
```diff
@@ -77,19 +78,20 @@ public CommandResult execute(Model model) throws CommandException {
      */
     private static Person createMarkedPerson(Person personToMark, Set<MonthPaid> monthPaid) {
         assert personToMark != null;
-
-        // TODO: should we use editPersonDescriptor here instead?
```
Comment (sumomomomomo) [inline src/main/java/seedu/address/logic/commands/MarkPaidCommand.java:81]: I think it's better to not delete TODO comments for now

File: src/main/java/seedu/address/model/person/Person.java [hunk 7]
```diff
@@ -71,8 +73,8 @@ public ClassId getClassId() {
      * Returns an immutable monthPaid set, which throws {@code UnsupportedOperationException}
```
Comment (sumomomomomo) [inline src/main/java/seedu/address/model/person/Person.java:73]:
```suggestion
     * Returns an immutable monthPaid sorted set, which throws {@code UnsupportedOperationException}
```

**Review by sumomomomomo — COMMENTED** [review]
Overall good, some nits

File: src/main/java/seedu/address/logic/Messages.java (hunk 5 above)
Comment (waihin26) [reply src/main/java/seedu/address/logic/Messages.java:57]: Using builder:append wont append the " ". itll result in the UI printing all the months paid together. Ie. 2024-022023-022018-03

Comment (sumomomomomo) [reply src/main/java/seedu/address/logic/Messages.java:57]: Suggest you conform to `builder::append` and change toString of `monthPaid` instead, similar to implementation for Tag

**Review by hooyushing — COMMENTED** [review]
LGTM

File: src/main/java/seedu/address/model/person/MonthPaid.java [hunk 8]
```diff
@@ -17,7 +17,7 @@ public MonthPaid(String monthsPaid) {
 
     @Override
     public String toString() {
-        return value;
+        return value + " ";
```
Comment (sumomomomomo) [inline src/main/java/seedu/address/model/person/MonthPaid.java:20]:
```suggestion
        return '[' + value + ']';
```

**Pull request merged**